tests/format/test_FormatSMVADSC.py

```diff
@@ -0,0 +1,20 @@
+import os
+
+import dxtbx
+
+
+def test_noninteger_pedestal(dials_regression, tmpdir):
+    filename = os.path.join(
+        dials_regression, "image_examples/APS_14BMC/q315_1_001.img",
+    )
+    # Read this file in as data
+    with open(filename, "rb") as f:
+        data = f.read()
+
+    # Write out with an inserted header item of a noninteger pedestal
+    test_file = tmpdir / "test_pedestal_001.img"
+    with test_file.open("wb") as f:
+        f.write(data.replace(b"DIM=2;\n", b"DIM=2;\nIMAGE_PEDESTAL=42.0;\n"))
```

Comment:
would suggest changing the `tmpdir` fixture to `tmp_path` and

```suggestion
    test_file = tmp_path / "test_pedestal_001.img"
    test_file.write_bytes(data.replace(b"DIM=2;\n", b"DIM=2;\nIMAGE_PEDESTAL=42.0;\n"))
```

Reply:
discussed this with Nick. Can leave this until we drop Python 2.7 support. After that point there should hopefully be no excuse to introduce new `py.path`s 🙂